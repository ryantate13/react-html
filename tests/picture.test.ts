import {picture} from '../src';

describe('picture', () => {
    it('renders with children', () => {
        expect(picture({}, 'TEST')).toMatchSnapshot();
    });
    
    it('renders arrays of children', () => {
        expect(picture({}, Array(10).fill('TEST'))).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(picture({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(picture()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(picture({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(picture({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(picture({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(picture({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(picture({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(picture({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(picture({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(picture({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(picture({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(picture({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(picture({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(picture({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(picture({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(picture({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(picture({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden=true', () => {
        expect(picture({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden=false', () => {
        expect(picture({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(picture({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(picture({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(picture({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(picture({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(picture({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(picture({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(picture({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(picture({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(picture({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(picture({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(picture({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(picture({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(picture({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope=true', () => {
        expect(picture({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope=false', () => {
        expect(picture({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(picture({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(picture({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(picture({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(picture({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(picture({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(picture({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style={"color":"#000000"}', () => {
        expect(picture({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex=1', () => {
        expect(picture({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(picture({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(picture({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(picture({translate: "no"})).toMatchSnapshot();
    });
});