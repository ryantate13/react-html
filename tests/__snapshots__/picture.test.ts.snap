// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`picture renders arrays of children 1`] = `
<picture>
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
</picture>
`;

exports[`picture renders with accessKey="a,b,c,d" 1`] = `
<picture
  accessKey="a,b,c,d"
/>
`;

exports[`picture renders with autoCapitalize="characters" 1`] = `
<picture
  autoCapitalize="characters"
/>
`;

exports[`picture renders with autoCapitalize="none" 1`] = `
<picture
  autoCapitalize="none"
/>
`;

exports[`picture renders with autoCapitalize="off" 1`] = `
<picture
  autoCapitalize="off"
/>
`;

exports[`picture renders with autoCapitalize="on" 1`] = `
<picture
  autoCapitalize="on"
/>
`;

exports[`picture renders with autoCapitalize="sentences" 1`] = `
<picture
  autoCapitalize="sentences"
/>
`;

exports[`picture renders with autoCapitalize="words" 1`] = `
<picture
  autoCapitalize="words"
/>
`;

exports[`picture renders with children 1`] = `
<picture>
  TEST
</picture>
`;

exports[`picture renders with className="foo bar baz" 1`] = `
<picture
  className="foo bar baz"
/>
`;

exports[`picture renders with contentEditable="false" 1`] = `
<picture
  contentEditable="false"
/>
`;

exports[`picture renders with contentEditable="true" 1`] = `
<picture
  contentEditable="true"
/>
`;

exports[`picture renders with dir="auto" 1`] = `
<picture
  dir="auto"
/>
`;

exports[`picture renders with dir="ltr" 1`] = `
<picture
  dir="ltr"
/>
`;

exports[`picture renders with dir="rtl" 1`] = `
<picture
  dir="rtl"
/>
`;

exports[`picture renders with draggable="false" 1`] = `
<picture
  draggable="false"
/>
`;

exports[`picture renders with draggable="true" 1`] = `
<picture
  draggable="true"
/>
`;

exports[`picture renders with empty attributes 1`] = `<picture />`;

exports[`picture renders with hidden=false 1`] = `
<picture
  hidden={false}
/>
`;

exports[`picture renders with hidden=true 1`] = `
<picture
  hidden={true}
/>
`;

exports[`picture renders with id="TEST" 1`] = `
<picture
  id="TEST"
/>
`;

exports[`picture renders with inputMode="decimal" 1`] = `
<picture
  inputMode="decimal"
/>
`;

exports[`picture renders with inputMode="email" 1`] = `
<picture
  inputMode="email"
/>
`;

exports[`picture renders with inputMode="none" 1`] = `
<picture
  inputMode="none"
/>
`;

exports[`picture renders with inputMode="numeric" 1`] = `
<picture
  inputMode="numeric"
/>
`;

exports[`picture renders with inputMode="search" 1`] = `
<picture
  inputMode="search"
/>
`;

exports[`picture renders with inputMode="tel" 1`] = `
<picture
  inputMode="tel"
/>
`;

exports[`picture renders with inputMode="text" 1`] = `
<picture
  inputMode="text"
/>
`;

exports[`picture renders with inputMode="url" 1`] = `
<picture
  inputMode="url"
/>
`;

exports[`picture renders with is="foo" 1`] = `
<picture
  is="foo"
/>
`;

exports[`picture renders with itemID="https://google.com/" 1`] = `
<picture
  itemID="https://google.com/"
/>
`;

exports[`picture renders with itemProp="https://google.com/" 1`] = `
<picture
  itemProp="https://google.com/"
/>
`;

exports[`picture renders with itemRef="foo bar" 1`] = `
<picture
  itemRef="foo bar"
/>
`;

exports[`picture renders with itemScope=false 1`] = `
<picture
  itemScope={false}
/>
`;

exports[`picture renders with itemScope=true 1`] = `
<picture
  itemScope={true}
/>
`;

exports[`picture renders with itemType="https://google.com/" 1`] = `
<picture
  itemType="https://google.com/"
/>
`;

exports[`picture renders with lang="" 1`] = `
<picture
  lang=""
/>
`;

exports[`picture renders with lang="es-419" 1`] = `
<picture
  lang="es-419"
/>
`;

exports[`picture renders with slot="TEST" 1`] = `
<picture
  slot="TEST"
/>
`;

exports[`picture renders with spellCheck="false" 1`] = `
<picture
  spellCheck="false"
/>
`;

exports[`picture renders with spellCheck="true" 1`] = `
<picture
  spellCheck="true"
/>
`;

exports[`picture renders with style={"color":"#000000"} 1`] = `
<picture
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`picture renders with tabIndex=1 1`] = `
<picture
  tabIndex={1}
/>
`;

exports[`picture renders with title="TEST" 1`] = `
<picture
  title="TEST"
/>
`;

exports[`picture renders with translate="no" 1`] = `
<picture
  translate="no"
/>
`;

exports[`picture renders with translate="yes" 1`] = `
<picture
  translate="yes"
/>
`;

exports[`picture renders without attributes 1`] = `<picture />`;
